src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,138 @@
+import {
+    Component,
+    Input,
+    ViewEncapsulation,
+    ContentChildren,
+    QueryList,
+    Renderer2,
+    ElementRef,
+    Optional,
+    SkipSelf,
+    ContentChild,
+    Output,
+    AfterViewInit,
+  HostListener
+} from '@angular/core';
+import { Router, RouterLinkActive, NavigationEnd, RouterState, ActivatedRoute, UrlTree } from '@angular/router';
+import 'rxjs/add/operator/filter';
+
+@Component({
+    selector: '[ux-navigation-item]',
+    templateUrl: './navigation-item.component.html',
+    host: {
+        '[class.active]': 'active',
+        '[class.selected]': 'expanded',
+    },
+    encapsulation: ViewEncapsulation.None
+})
+export class NavigationItemComponent implements AfterViewInit {
+    @Input() header: string;
+    @Input() icon: string;
+    @Input() expanded: boolean = false;
+    @Input() link: string;
+
+    @Output()
```

Comment:
Can remove @Output as this is not an eventemitter. Also likely dont need _level any more, just make this:

```typescript
level: number = 1;
```

Remove the @output from active as well. Also active getter never uses _active so this can be removed

Reply:
@Output removed from active().
level now a public member. level() removed.